server/tilesource/mapniksource.py

```diff
@@ -91,7 +91,11 @@ def __init__(self, path, projection=None, style=None, **kwargs):
             # -180 to +180 degrees horizontally.  We should accept this as a
             # parameter eventually.
             inProj = pyproj.Proj('+init=epsg:4326')
-            outProj = pyproj.Proj(self.projection)
+            if isinstance(self.projection, six.binary_type):
+                projection = self.projection.decode('utf8')
+            else:
+                projection = self.projection
```

Comment:
We explicitly `encode` the projection to a binary string earlier, but here we are explicitly converting to a unicode string.  It looks like `pyproj.Proj` always handles unicode (in Python 2 and 3), but fails with bytes in Python 3.  So, safely, since we converted to bytes earlier, here we can safely say `outProj = pyproj.Proj(self.projection.decode('utf8'))`.

Reply:
Though a comment might be warranted to say why that is safe...